src/Components/LaunchDescriptionList/index.js

```diff
@@ -5,23 +5,32 @@ import { ExpandableSection, DescriptionList, DescriptionListTerm, DescriptionLis
 import { useSourcesData } from '../Common/Hooks/sources';
 import { useWizardContext } from '../Common/WizardContext';
 import { instanceType, region } from '../ProvisioningWizard/steps/ReservationProgress/helpers';
+import { useQuery } from 'react-query';
+import { fetchLaunchTemplates } from '../../API';
+import { humanizeProvider } from '../Common/helpers';
 
 const LaunchDescriptionList = ({ imageName }) => {
-  const [{ chosenRegion, chosenSshKeyName, uploadedKey, chosenInstanceType, chosenNumOfInstances, chosenSource, sshPublicName, provider }] =
-    useWizardContext();
+  const [
+    { chosenRegion, chosenSshKeyName, uploadedKey, chosenInstanceType, chosenNumOfInstances, chosenSource, sshPublicName, provider, chosenTemplate },
+  ] = useWizardContext();
   const { sources } = useSourcesData(provider);
+  const { data: templates } = useQuery(['Templates', `${chosenRegion}-${chosenSource}`], () => fetchLaunchTemplates(chosenSource, chosenRegion), {
+    enabled: !!chosenSource && !!chosenRegion,
+  });
+
   const [isExpanded, setIsExpanded] = React.useState(true);
   const onToggle = (isExpanded) => {
     setIsExpanded(isExpanded);
   };
 
   const getChosenSourceName = () => sources?.find((source) => source.id === chosenSource).name;
+  const templateName = templates.find((template) => template.id === chosenTemplate)?.name;
```

Comment:
`templates` might be null I'm guessing (in case the query is disabled or there was some error, or it's still loading